rotkehlchen/globaldb/upgrades/v4_v5.py

```diff
@@ -58,6 +58,26 @@ def _reset_curve_cache(write_cursor: 'DBCursor') -> None:
     write_cursor.execute('DELETE FROM general_cache WHERE key LIKE "%CURVE%"')
 
 
+def _remove_name_from_contracts(write_cursor: 'DBCursor') -> None:
+    """Removes name column from contract_data table"""
+    write_cursor.execute("""
+    CREATE TABLE IF NOT EXISTS contract_data_new (
+        address VARCHAR[42] NOT NULL,
+        chain_id INTEGER NOT NULL,
+        abi INTEGER NOT NULL,
+        deployed_block INTEGER,
+        FOREIGN KEY(abi) REFERENCES contract_abi(id) ON UPDATE CASCADE ON DELETE SET NULL,
+        PRIMARY KEY(address, chain_id)
+    );
+    """)
+    write_cursor.execute(
```

Comment:
It has already happened that we dropped support for an old update and it broke the logic. We can follow here a similar approach to what left fixed in https://github.com/rotki/rotki/pull/5872/files

Reply:
With address book the issue was that there was an assumption that the table was created with `DB_SCRIPT_CREATE_TABLES`. In case of `contract_data`,  it is explicitly created in v3_v4 upgrade https://github.com/rotki/rotki/blob/378b4458291ef457b964a62ee464d4dd35c220b0/rotkehlchen/globaldb/upgrades/v3_v4.py#L119, so we can be sure that the table exists.

Reply:
No please. Do 100% future-proof upgrades. No assumptions. No "can be sure".

Reply:
Okay